packages/content-source-maps/src/rest/encodeCPAResponse.ts

```diff
@@ -0,0 +1,167 @@
+import { has, get, set } from 'json-pointer';
+import { SourceMapMetadata, combine } from '../encode.js';
+import { CPAEntry, CPAEntryCollection, EditorInterfaceSource, FieldType } from '../types.js';
+import { clone, getHref, isBuiltinNamespace, isSupportedWidget } from '../utils.js';
+
+const applyEncoding = (
+  item: CPAEntry,
+  fieldTypes: FieldType[],
+  editorInterfaces: EditorInterfaceSource[],
+  targetOrigin?: 'https://app.contentful.com' | 'https://app.eu.contentful.com',
+) => {
+  if (!item.fields) {
+    return;
+  }
+  const { contentSourceMaps } = item.sys;
+  if (!contentSourceMaps) {
+    console.error('Content source maps data is missing');
+    return;
+  }
+  const { mappings } = contentSourceMaps;
+
+  for (const pointer in mappings) {
+    const { source } = mappings[pointer];
+    const space = item.sys.space.sys.id;
+    const environment = item.sys.environment.sys.id;
+    const entityId = item.sys.id;
+    const entityType = item.sys.type;
+    const fieldType = fieldTypes[source.fieldType];
+    const editorInterface = editorInterfaces[source.editorInterface];
+
+    // Skip unsupported widgets
+    if (
+      isBuiltinNamespace(editorInterface.widgetNamespace) &&
+      !isSupportedWidget(editorInterface.widgetId)
+    ) {
+      continue;
+    }
+
+    if (has(item, pointer)) {
+      const currentValue = get(item, pointer);
+      if (currentValue === null) {
+        return;
+      }
+      const fieldParts = pointer.split('/'); // Split the pointer into parts
+      const field = fieldParts.pop(); // Get the last part, which is the field name
+      if (!field) {
+        console.error('Field name could not be extracted from the pointer', pointer);
+        return;
+      }
+      // Determine the locale: if 'currentValue' is an object, each key represents a different locale
+      // Otherwise, the single locale is directly available under 'item.sys.locale'
+      if (typeof currentValue === 'object') {
+        const locales = Object.keys(currentValue);
+        locales.forEach((locale) => {
+          const href = getHref(
+            entityId,
+            entityType,
+            space,
+            environment,
+            field,
+            locale,
+            targetOrigin,
+          );
+          const hiddenStrings: SourceMapMetadata = {
```

Comment:
we could extract this too and instead of calling getHref we call createSourceMapMetadata, then we can share this between REST and GQL too 🤔